deps/build_OpenBLAS.jl

```diff
@@ -0,0 +1,46 @@
+using BinaryProvider # requires BinaryProvider 0.3.0 or later
```

Comment:
if this PR is about `gels`, why there is draft build script here?

Reply:
> @JerryChen97 gels should have a gpu interface in MAGMA right? why this only binds the CPU?

I am still trying to solve a bug related with the sgels' gpu interface in MAGMA.

Reply:
> if this PR is about `gels`, why there is draft build script here?

I forgot to delete them...

Reply:
So this PR is not ready for review yet? OK, it'd be better to add WIP in the title if this is not ready for review.